src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -5,72 +5,61 @@
  */
 export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
     constructor(
         private platformInfo: PlatformInformation,
-        /**
-         * This is a function instead of a string because the server path can change while the extension is active (when the option changes).
-         */
-        private getServerPath: (platform: PlatformInformation) => string,
         private channel: vscode.OutputChannel,
         private extensionPath: string
     ) {}
 
     private hostInfo: HostExecutableInformation | undefined;
 
     async getHostExecutableInfo(): Promise<HostExecutableInformation> {
-        let dotnetRuntimePath = commonOptions.dotnetPath;
-        const serverPath = this.getServerPath(this.platformInfo);
-
-        // Check if we can find a valid dotnet from dotnet --version on the PATH.
-        if (!dotnetRuntimePath) {
-            const dotnetPath = await this.findDotnetFromPath();
-            if (dotnetPath) {
-                return {
-                    version: '' /* We don't need to know the version - we've already verified its high enough */,
-                    path: dotnetPath,
-                    env: this.getEnvironmentVariables(dotnetPath),
-                };
+        let dotnetExecutablePath: string;
+        if (commonOptions.dotnetPath) {
+            const dotnetExecutableName = this.getDotnetExecutableName();
+            dotnetExecutablePath = path.join(commonOptions.dotnetPath, dotnetExecutableName);
+        } else {
+            if (this.hostInfo) {
+                return this.hostInfo;
             }
-        }
-
-        // We didn't find it on the path, see if we can install the correct runtime using the runtime extension.
-        if (!dotnetRuntimePath) {
-            const dotnetInfo = await this.acquireDotNetProcessDependencies(serverPath);
-            dotnetRuntimePath = path.dirname(dotnetInfo.path);
-        }
 
-        const dotnetExecutableName = this.getDotnetExecutableName();
-        const dotnetExecutablePath = path.join(dotnetRuntimePath, dotnetExecutableName);
-        if (!existsSync(dotnetExecutablePath)) {
-            throw new Error(`Cannot find dotnet path '${dotnetExecutablePath}'`);
+            this.channel.appendLine(`Acquiring .NET runtime version ${DotNetRuntimeVersion}`);
+            const extensionArchitecture = (await this.getArchitectureFromTargetPlatform()) ?? process.arch;
+            const findPathRequest: IDotnetFindPathContext = {
+                acquireContext: {
+                    version: DotNetRuntimeVersion,
+                    requestingExtensionId: CSharpExtensionId,
+                    architecture: extensionArchitecture,
+                    mode: 'runtime',
+                },
+                versionSpecRequirement: 'greater_than_or_equal',
+            };
+            const result = await vscode.commands.executeCommand<string>('dotnet.findPath', findPathRequest);
```

Comment:
fyi @nagilson - https://github.com/dotnet/vscode-dotnet-runtime/blob/a2e6a5c385190fe8e48e51dca17897b3a209d4f1/vscode-dotnet-runtime-extension/src/test/functional/DotnetCoreAcquisitionExtension.test.ts#L235 appears to indicate it returns a `IDotnetAcquireResult`, but it appears to only ever return a string

Reply:
It can return an `IDotnetAcquireResult` if the `existingDotnetPath` is set. It should always return an `IDotnetAcquireResult`. This was a mistake on my part, thank you for bringing this to my attention. I will make a fix.